docs/input.md

```diff
@@ -90,20 +90,199 @@ def setup(): # Only drawn once, so just using setup()
 
 which results in:
 
-![Write Name](img/write_name.png)
+![Write Name](img/write_name.png){: loading=lazy }
 
 
-*Ask the user for a radius. We need to convert the string to an integer, we can do this using int(). Draw a circle of that radius at (100, 100):*
+*Ask the user for a diameter. We need to convert the string to an integer, we can do this using int(). Draw a circle of that diameter at (100, 100):*
 
-```python
+```python hl_lines="5"
 %%ignite
 
 def setup(): # Only drawn once, so just using setup()
     size(200, 200)
-    radius = int(input("What radius should the circle be?: ")) # cast the string to an integer
-    fill_circle(100, 100 radius)
+    diameter = int(input("What diameter should the circle be?: ")) # cast the string to an integer
+    fill_circle(100, 100 diameter)
 ```
 
 which results in:
 
-![Ask circle radius](img/circle_radius.png)
+![Ask circle diameter](img/circle_radius.png){: loading=lazy }
+
+### Check for key presses
+
+Spark features multiple ways to check for user key presses:
+
+| Name                            | Type                  | Description                                                  |
+|---------------------------------|-----------------------|--------------------------------------------------------------|
+| [key](#key)                     | Variable              | A string of what key was last pressed                        |
+| [keys_held()](#keys_held)       | Function              | A function that returns True if the provided key is held     |
+| [key_pressed()](#key_pressed)   | User defined function | A definable function that activates when any key is pressed  | 
+| [key_released()](#key_released) | User defined function | A definable function that activates when any key is released |
+| [key_repeated()](#key_repeated) | User defined function | A definable function that activates when any key is held     |
+
+#### key
+
+A string that is the last key pressed
```

Comment:
The string corresponds to the last key that triggered an event (i.e. pressed, repeated, or released), not just pressed. This behaviour was chosen to mimic Processing, and to let the user write the `key_released` and `key_repeated` functions just using the `key` variable

Reply:
Although this is true, considering the audience I think the wording of pressed may be a bit clearer as necessarily for it to be repeated or released it must first be pressed anyway, but in common parlance pressing a key is better understood than repeating or releasing

Reply:
The only reason I would want to clarify is that in processing, people would do
```java
if (keyPressed){
    if (key == LEFT) {
        // Go left
    } else if (key == RIGHT) {
        // Go right
   }
} 
```
and if you press left, then press right, then release left, you'd end up moving right even though you're only holding left, and that was when we explained how it worked. The new functionality makes it harder to do that, but that was something I saw students run into

Reply:
Hmm, ok I will adjust it since that's a fair point

Reply:
How does 
> A string that is the last key event (a key being pressed, released, or held down). So for example if you press, hold or release ++shift++ then `#!python key == "Shift"`.

sound

Reply:
That sounds fine! If you wanted to go even less technical, you could say it's the last key that "did something", where the something is be pressed, be released, or be held, but I'm fine with your phrasing there